packages/wonder-blocks-dropdown/components/action-menu.js

```diff
@@ -109,69 +126,153 @@ export default class ActionMenu extends React.Component<Props, State> {
         };
     }
 
-    /**
-     * Update the internal state and notify the parent component when menu is
-     * toggled
-     */
-    handleToggleMenu = (opened: boolean) => {
+    handleItemSelected = () => {
+        // close menu
+        this.handleOpenChanged(false);
+
+        // Bring focus back to the opener element.
+        if (this.openerElement) {
+            this.openerElement.focus();
+        }
+    };
+
+    handleOpenChanged = (opened: boolean, keyboard?: boolean) => {
         this.setState({
-            opened: opened,
+            opened,
+            keyboard,
         });
 
         if (this.props.onToggle) {
             this.props.onToggle(opened);
         }
     };
 
+    handleOptionSelected = (selectedValue: string) => {
+        const {onChange, selectedValues} = this.props;
+
+        // If either of these are not defined, return.
+        if (!onChange || !selectedValues) {
+            return;
+        }
+
+        if (selectedValues.includes(selectedValue)) {
+            const index = selectedValues.indexOf(selectedValue);
+            const updatedSelection = [
+                ...selectedValues.slice(0, index),
+                ...selectedValues.slice(index + 1),
+            ];
+            onChange(updatedSelection);
+        } else {
+            // Item was newly selected
+            onChange([...selectedValues, selectedValue]);
+        }
+        this.handleItemSelected();
+    };
+
+    getMenuItems(): Array<DropdownItem> {
```

Comment:
Is this method copied over from the `Dropdown` component?

Reply:
Yes, now `ActionMenu` does with `Dropdown` was doing, so I decided to copy this logic from `dropdown.js`.